### ISSUE TITLE
Redirect history entries not deleted 

### ISSUE THREAD

**Issue by anewuser** [igorlogius/open-in-temp-container#7]
Hello. Some URLs aren't being fully deleted from my browsing history. This seems to happen, for instance, when there's a redirection involved.

**To Reproduce**
1. Allow the extension to access your browsing history
2. Set the extension button to open a new private tab
3. Go to https://old.reddit.com/r/example or https://old.reddit.com/r/ExAmPLe/
4. Close the tab
5. Start typing out `old` into the address bar or open your browsing history (the Firefox Library window) and search for `old`

https://github.com/igorlogius/open-in-temp-container/assets/6733770/1ce41f74-f5f0-45cd-bdc7-539bc3987005

Windows 10
Firefox 122.0.1

**Comment by igorlogius** [comment]
Thanks for reporting this issue.

I am sadly not able to reproduce the issue (yet). 
When i follow your steps, this is how it looks for me, with a new unmodified profile.

https://github.com/igorlogius/open-in-temp-container/assets/67047467/b24bcfa5-38be-45a7-bb35-4b6295fbfea9

Can you reproduce the issue on yourside with a 2nd new unmodified profile?

**Comment by igorlogius** [comment]
Ok, after some experimenting it seems the issue is not with the actual history but might be realated to some kind of caching done by the search element in the library view.

So ... the history elements should not exist anymore ... but because of the search that has been done once, they have been cached ... not sure how long those searches persist ... but i suspect it to not be to long... but i am no expert here. 

So tldr; this is an issue with the browser history search and not really with the addon. Not much i can do here. There is AFAIK no way for me to influence this behaviour. 

You could try open a bug on bugzilla ... but it might just be tradehoff so that searches can be quickly reapplied. - just guessing, but thats would i'd guess from what i am seeing.

**Comment by anewuser** [comment]
>because of the search that has been done once

This isn't caused by previous searches. It happens even if you've never opened the library before, and those URLs persist as address bar suggestions too.  

Note that the examples I've mentioned redirect to other URLs:

https://old.reddit.com/r/example -> `https://old.reddit.com/r/example/` (a trailing slash is added)
https://old.reddit.com/r/ExAmPLe/ -> `https://old.reddit.com/r/example/` (different capitalization)

The destination URLs are completely deleted, but the original URLs are preserved as address bar suggestions/"ghost" history entries. 

If you still think this is a Firefox bug, I'm going to report it to Mozilla.

**Comment by igorlogius** [comment]
> This isn't caused by previous searches. It happens even if you've never opened the library before, and those URLs persist as address bar suggestions too.

mmh, ok ... maybe its something else then ... those redirect ... 
maybe there is something here ... let me think about this a little.

**Comment by igorlogius** [comment]
Sidenote: I can not confirm that the redirect URLs persist in the libarary history. Do they in your case? In your video they only seem to appear when you search for `old`.

https://github.com/igorlogius/open-in-temp-container/assets/67047467/b2196523-f1fd-4d4a-bd25-4f5740aa64f0

**Comment by anewuser** [comment]
> In your video they only seem to appear when you search for `old`.

Yes, I called them "ghost entries" because they only appear as address bar suggestions or history search results.

**Comment by igorlogius** [comment]
> "ghost entries"

spooky! I like it. :-D 

Sadly this doesnt seem like something i can influence or at least i  dont know how.
Maybe it would be best to ask about this behaviour in mozillas addon matrix channel and see if anyone has any insight on this before opening a bug report on bugzilla.

**Comment by igorlogius** [comment]
I think i just noticed something interesting ... i think the ghost entries might have something todo with the fact that the Temp container tab in which the navigation happend is still open/exists. Not 100% sure yet, but it seems like the ghost entries aren't there anymore when i remove the tab before the deletions activate.

**Comment by igorlogius** [comment]
after some more experimentation ...  it seems like i was mistaken ... i am again now sure why and how these entries persist. In my opinion that is a bug or at least something undesirable, when the delete history was explicitely called.

**Comment by anewuser** [comment]
I've filed an issue on Bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1880883

**Comment by igorlogius** [comment]
i wouldn't get my hopes up ... since this is an issue with an addon i don't think it will get a very hight priority, so it might take while for anyone to take a look.

**Comment by igorlogius** [comment]
Closing this for now, since there isnt really anything i could do at the moment.

**Comment by anewuser** [comment]
The Bugzilla issue has been marked as invalid: https://bugzilla.mozilla.org/show_bug.cgi?id=1880883#c3

**Comment by igorlogius** [comment]
Ok, from Rob Wu's comment i guess i am doing something wrong ... i guess i was mistaken in thinking that the redirects would be picked up by the onUpdated event ... guess i have to use the webRequest or webNavigation hooks instead for that ... i'll take another look.

**Comment by igorlogius** [comment]
Seems to work now, as far as i can tell

https://github.com/igorlogius/open-in-temp-container/assets/67047467/972055d1-02ea-4c9f-8460-6c19b625154d

**Comment by igorlogius** [comment]
@anewuser Please check if the issue has been resolved with version `1.1.19` for you too now.

**Comment by anewuser** [comment]
After accepting the new required permission and updating the extension, I had the opposite issue: pre-redirection URLs were being deleted, while target URLs were being kept. I think that I had the "deletion delay" set to a value shorter than 3000, but I'm not sure.

Anyway, reinstalling the extension seems to have fixed this. It now deletes all URLs accordingly. Thank you!
